src/main/java/org/testng/internal/Invoker.java

```diff
@@ -835,6 +836,9 @@ int retryFailed(Object instance,
 
       ParameterBag bag = handler.createParameters(tm, parameters, allParameters, testContext);
       Object[] parameterValues = Parameters.getParametersFromIndex(bag.parameterHolder.parameters, parametersIndex);
+      if (bag.parameterHolder.origin == ParameterHolder.ParameterOrigin.NATIVE) {
```

Comment:
I have some doubts about this line. Could you add a test that mix origins (native and one and/or 2 others origins)?

Reply:
@juherr you doubts are valid. But the functionality would have been broken even without this line (that's my calculated guess) because the mode basically evaluates to just one value (be it with my change or without my change), but you are asking for a combination of more than one modes. So if it breaks, my guess is, its an existing regression that should be out of the scope of this issue. WDYT ?

Reply:
Ok, I thought it was working before but we have no test that already mixes param origin.